api/client/src/main/java/org/projectnessie/client/auth/oauth2/JacksonSerializers.java

```diff
@@ -56,7 +56,9 @@ public SecondsToInstantDeserializer() {
     public Instant deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
       if (p.currentToken().isNumeric()) {
         int seconds = p.getValueAsInt();
-        return Instant.now().plusSeconds(seconds);
+        if (seconds != 0) {
+          return Instant.now().plusSeconds(seconds);
```

Comment:
This change LGTM as it stands, but in general, would it be preferable to avoid binding to the current clock (`Instant.now()`) in JSON deserializers? Could we delegate that to the code that uses this value?

Reply:
Agreed. I will tackle that in a follow-up PR.